src/pykeen/evaluation/rank_based_evaluator.py

```diff
@@ -300,12 +351,128 @@ def finalize(self) -> RankBasedMetricResults:  # noqa: D102
         )
+        if not self.clear_on_finalize:
+            return result
+
         # Clear buffers
         self.ranks.clear()
         self.num_candidates.clear()
 
         return result
 
+    def finalize_multi(self, num: int = 1_000, seed: int = 42) -> Mapping[str, Sequence[float]]:
+        """Bootstrap from :meth:`finalize`.
+
+        :param num:
+            the number of resampling steps
+        :param seed:
+            the random seed.
+
+        :return:
+            a flat dictionary from metric names to list of values
+        """
+        result: DefaultDict[str, List[float]] = defaultdict(list)
+
+        for i in range(num):
+            rank_and_candidates = _iter_ranks(ranks=self.ranks, num_candidates=self.num_candidates)
+            rank_and_candidates = map(functools.partial(RankPack.resample, seed=seed + i), rank_and_candidates)
+            single_result = RankBasedMetricResults.from_ranks(
+                metrics=self.metrics, rank_and_candidates=rank_and_candidates
+            )
+            for k, v in single_result.to_flat_dict().items():
+                result[k].append(v)
+        return result
+
+    def finalize_with_confidence(
+        self,
+        estimator: Union[str, Callable[[Sequence[float]], float]] = np.median,
+        ci: Union[int, str, Callable[[Sequence[float]], float]] = 90,
+        n_boot: int = 1_000,
+        seed: int = 42,
+    ) -> Mapping[str, Tuple[float, float]]:
+        """Finalize result with confidence estimation via bootstrapping.
+
+        Example
+        -------
+        Start by training a model (here, only for a one epochs)
+
+        >>> from pykeen.pipeline import pipeline
+        >>> result = pipeline(dataset="nations", model="rotate", training_kwargs=dict(num_epochs=1))
+
+        Create an evaluator with `clear_on_finalize` set to `False`, e.g., via
+
+        >>> from pykeen.evaluation import evaluator_resolver
+        >>> evaluator = evaluator_resolver.make("rankbased", clear_on_finalize=False)
+
+        Evaluate *once*, this time ignoring the result
+
+        >>> evaluator.evaluate(model=result.model, mapped_triples=result.training.mapped_triples)
+
+        Now, call `finalize_with_confidence` to obtain estimates for metrics together with confidence intervals
+
+        >>> evaluator.finalize_with_confidence(n_boot=10)
+
+        :param estimator:
+            the estimator of central tendency.
+        :param ci:
+            the confidence interval
+        :param n_boot:
+            the number of resamplings to use for bootstrapping
+        :param seed:
+            the random seed
+
+        :return:
+            a dictionary from metric names to (central tendency, confidence) pairs
+        """
+        return {
+            k: summarize_values(vs, estimator=estimator, ci=ci)
+            for k, vs in self.finalize_multi(num=n_boot, seed=seed).items()
```

Comment:
I'd like to see the argument `num` and `n_boot` be the same.

Reply:
https://github.com/pykeen/pykeen/pull/1084/commits/604f77f4893a900ff80c3be213cbc406e6267eaa